### PR TITLE
Remove Roslyn from the shared framework

### PR DESCRIPTION
**Pull request by natemcmaster**
Reacting to https://github.com/aspnet/Mvc/pull/8755.

This is one more step towards resolving #4004. This removes Roslyn from the shared framework.

### CODE REVIEW

File: build/artifacts.props [hunk 1]
```diff
@@ -44,6 +44,7 @@
     <PackageArtifact Include="Microsoft.AspNetCore.MiddlewareAnalysis" Category="ship" />
     <PackageArtifact Include="Microsoft.AspNetCore.Mvc.Analyzers" Category="ship" />
     <PackageArtifact Include="Microsoft.AspNetCore.Mvc.Api.Analyzers" Category="ship" />
+    <PackageArtifact Include="Microsoft.AspNetCore.Mvc.Razor.Extensions" Category="ship" />
```
Comment (rynowak) [inline build/artifacts.props]: FYI this package will be in the no-mans-land category of "ships as a package but not in the shared framework"

If you're doing this as a temporary solution for now, or if this is the correct marking then that's fine.

Comment (natemcmaster) [reply build/artifacts.props]: This is what this marking means, tell me if it's right:

* Not in the shared framework
* Should ship to nuget.org

Comment (rynowak) [reply build/artifacts.props]: OK, cool beans!

**Comment by natemcmaster** [comment]
Tests are failing because the nuspec wasn't updated. This PR is blocked on getting https://github.com/aspnet/Mvc/pull/8758 merged.

**Review by pranavkm — APPROVED** [review]

**Pull request merged**